ESP8266/ESP8266.cpp

```diff
@@ -291,30 +294,40 @@ bool ESP8266::close(int id)
 
 void ESP8266::setTimeout(uint32_t timeout_ms)
 {
-    _parser.setTimeout(timeout_ms);
+    _parser.set_timeout(timeout_ms);
 }
 
 bool ESP8266::readable()
 {
-    return _serial.readable();
+    if ( _serial.poll( POLLIN ) & POLLIN ) return true;
+    return false;
```

Comment:
couldn't this be `return _serial.readable()`?

Reply:
Hi @geky  UARTSerial does private inheritance of SerialBase and prevents us from calling readable/writable. In addition, I think poll call implementation is little better in that it avoids spinning.

Reply:
You would need to be calling `UARTSerial::readable()`, not `SerialBase::readable`. You need to know if there's anything in the buffer, not the hardware.

Presumably you're avoiding https://github.com/ARMmbed/mbed-os/pull/5066? That could otherwise be avoided with by saying `_serial.FileHandle::readable()`.

I don't believe these functions are ever used anyway?

Reply:
See your point, yes we want to check the buffer and not the hardware fifo/buffer, I'll change it to using _serial.FileHandle::readable() and serial.FileHandle::writable().